internal/exec/util/path_sort.go

```diff
@@ -20,7 +20,7 @@ import (
 
 func Depth(path string) uint {
 	var count uint = 0
-	for p := filepath.Clean(path); p != "/"; count++ {
+	for p := filepath.Clean(path); p != "/" && p != "."; count++ {
```

Comment:
What happens here if someone provides `./subpath/root`?

Reply:
I guess in that case we do want to calculate depth. The only cases where we don't are `/` and `.`?

Should we special case and error for `..`?

Reply:
We always want to calculate depth, we just need to terminate the loop once we reach steady state.  The goal is to produce a relative value for comparing the depths of two paths both in the same root, so the exact numerical cost of that root doesn't really matter.

`filepath.Dir("..")` returns `.`, so we don't need to special-case `..`.